src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,37 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH_DATA = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (findUserByLogin(user) != null) {
+            throw new InvalidDataException("User with this login has been already registered");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("Login mustn't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("Password mustn't be null");
+        }
+        if (user.getLogin().length() < MIN_LENGTH_DATA) {
+            throw new InvalidDataException("Login must contain at least "
+                    + MIN_LENGTH_DATA + " characters");
+        }
+        if (user.getPassword().length() < MIN_LENGTH_DATA) {
+            throw new InvalidDataException("Password must contain at least "
+                    + MIN_LENGTH_DATA + " characters");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new InvalidDataException("User's age must be at least " + MIN_AGE + " years old");
+        }
+        return storageDao.add(user);
+    }
+
+    private User findUserByLogin(User user) {
+        String login = user.getLogin();
+        return storageDao.get(login);
     }
```

Comment:
The method `findUserByLogin` is not covered by any tests. Ensure that you have tests that cover this method to check if it behaves as expected when finding a user by login.